core/src/main/java/com/amplifyframework/core/AmplifyConfiguration.java

```diff
@@ -53,81 +57,115 @@
      * @throws AmplifyException If the JSON does not represent a valid AmplifyConfiguration
+     * @deprecated This method is now deprecated in favor of {@link #builder(JSONObject)}.
      */
+    @Deprecated
     @NonNull
     public static AmplifyConfiguration fromJson(@NonNull JSONObject json) throws AmplifyException {
-        final List<CategoryConfiguration> possibleConfigs = Arrays.asList(
-            new AnalyticsCategoryConfiguration(),
-            new ApiCategoryConfiguration(),
-            new AuthCategoryConfiguration(),
-            new DataStoreCategoryConfiguration(),
-            new HubCategoryConfiguration(),
-            new LoggingCategoryConfiguration(),
-            new PredictionsCategoryConfiguration(),
-            new StorageCategoryConfiguration()
-        );
-
-        final Map<String, CategoryConfiguration> actualConfigs = new HashMap<>();
-        try {
-            for (CategoryConfiguration possibleConfig : possibleConfigs) {
-                String categoryJsonKey = possibleConfig.getCategoryType().getConfigurationKey();
-                if (json.has(categoryJsonKey)) {
-                    possibleConfig.populateFromJSON(json.getJSONObject(categoryJsonKey));
-                    actualConfigs.put(categoryJsonKey, possibleConfig);
-                }
-            }
-        } catch (JSONException error) {
-            throw new AmplifyException(
-                "Could not parse amplifyconfiguration.json ",
-                error, "Check any modifications made to the file."
-            );
-        }
-        return new AmplifyConfiguration(Immutable.of(actualConfigs));
+        return builder(json).build();
     }
 
     /**
      * Creates an AmplifyConfiguration from an amplifyconfiguration.json file.
+     *
      * @param context Context needed for reading JSON file
      * @return An Amplify configuration instance
      * @throws AmplifyException If there is a problem in the config file
+     * @deprecated This method is now deprecated in favor of
+     *          {@link AmplifyConfiguration#builder(Context)}.
      */
     @SuppressWarnings("WeakerAccess")
+    @Deprecated
     @NonNull
     public static AmplifyConfiguration fromConfigFile(@NonNull Context context) throws AmplifyException {
-        return fromConfigFile(context, getConfigResourceId(context));
+        return builder(context, getConfigResourceId(context)).build();
     }
 
     /**
      * Creates an AmplifyConfiguration from a particular configuration file.
+     *
      * @param context Android Context
      * @param configFileResourceId
      *        The Android resource ID of a raw resource which contains
      *        an amplify configuration as JSON
      * @return An Amplify configuration instance
      * @throws AmplifyException If there is a problem in the config file
+     * @deprecated This method is now deprecated in favor of
+     *          {@link AmplifyConfiguration#builder(Context, int)}.
      */
+    @Deprecated
```

Comment:
Ditto, you have `builder(..., ...).build()` around the codebase, so why not just keep this and use it, since it's cleaner looking.